src/main/java/frc/robot/subsystems/LimelightNavigation.java

```diff
@@ -35,53 +43,125 @@ public class LimelightNavigation extends SubsystemBase {
+
+  public static Pigeon2 m_pigeon2;
+  boolean m_InitializeDFromTag = false;
+
+  double m_latency = 0.0;
+  double m_limelightSamplesCaptured = 0;
 
-  private Pigeon2 pigeon2;
   //private Pigeon2Configuration pigeon2Config;
 
   int x;
 
   /** Creates a new LimelightNavigation. */
   public LimelightNavigation(Map<String, RelativeEncoder> encoders) {
-    //BUGBUG: Get this in RobotContainer, pass to this subsystem.
-	Map<String, RelativeEncoder> Encoders = encoders;
-    m_FrontLeftWheel_Endocer	= Encoders.get("Front Left");
-    m_FrontRightWheel_Encoder	= Encoders.get("Front Right");
-    m_RearLeftWheel_Encoder		= Encoders.get("Rear Left");
-    m_RearRightWheel_Encoder	= Encoders.get("Rear Right");
+    // BUGBUG: Get this in RobotContainer, pass to this subsystem.
+    Map<String, RelativeEncoder> Encoders = encoders;
+    m_FrontLeftWheel_Endocer = Encoders.get("Front Left");
+    m_FrontRightWheel_Encoder = Encoders.get("Front Right");
+    m_RearLeftWheel_Encoder = Encoders.get("Rear Left");
+    m_RearRightWheel_Encoder = Encoders.get("Rear Right");
+
+
+    Pose2d initialPoseMeters = new Pose2d();
+    m_Kinematics = new MecanumDriveKinematics(Constants.Misc.FrontLeftDriveWheel_Position_Meters,
+        Constants.Misc.FrontRightDriveWheel_Position_Meters,
+        Constants.Misc.RearLeftDriveWheel_Position_Meters, Constants.Misc.RearRightDriveWheel_Position_Meters);
+    m_DrivePoseEstimator = new MecanumDrivePoseEstimator(m_Kinematics, new Rotation2d(0),
+        new MecanumDriveWheelPositions(
+            m_FrontLeftWheel_Endocer.getPosition()*Constants.Misc.metersPerTick, 
+            m_FrontRightWheel_Encoder.getPosition()*Constants.Misc.metersPerTick,
+            m_RearLeftWheel_Encoder.getPosition()*Constants.Misc.metersPerTick,
+             m_RearRightWheel_Encoder.getPosition()*Constants.Misc.metersPerTick),
+        initialPoseMeters);
+
+    m_pigeon2 = new Pigeon2(CanID.Pigeon2);
+    var pigeon2Config = new Pigeon2Configuration();
+  }
 
+  public static Rotation2d getPoseHeading() {
+    return m_DrivePoseEstimator.getEstimatedPosition().getRotation();
+  }
 
-    m_Kinematics	= new MecanumDriveKinematics(Constants.Misc.FrontLeftDriveWheel_Position_Meters, Constants.Misc.FrontRightDriveWheel_Position_Meters,
-		Constants.Misc.RearLeftDriveWheel_Position_Meters, Constants.Misc.RearRightDriveWheel_Position_Meters);
-    m_DriveOdometry	= new MecanumDriveOdometry(m_Kinematics, new Rotation2d(0), new MecanumDriveWheelPositions(
-    	m_FrontLeftWheel_Endocer.getPosition(), m_FrontRightWheel_Encoder.getPosition(), m_RearLeftWheel_Encoder.getPosition(), m_RearRightWheel_Encoder.getPosition()
-    ));
+  public static Rotation2d getGyroHeading() {
+    return m_pigeon2.getRotation2d();
+  }
 
-    pigeon2 = new Pigeon2(CanID.Pigeon2);
-    //pigeon2Config = new Pigeon2Configuration();
+  public void ledsOff() {
+    NetworkTableInstance.getDefault().getTable("limelight").getEntry("ledMode").setNumber(1);
```

Comment:
we should probably break these (and `ledsOn`) into a method and use constants for `1` and `2`

Reply:
Gotcha.